### ISSUE TITLE
Generates Python reserved words

### ISSUE THREAD

**Issue by scossu** [elijahr/python-autopxd2#3]
When processing https://github.com/srdja/Collections-C/blob/master/src/include/array.h#L110 , the following definition is created:

    cc_stat array_subarray(Array* ar, size_t from, size_t to, Array** out)

Cython throws an error at this line because `from` is a Python reserved word. Can a list of such words be maintained so those words are mangled in the pxd?

**Comment by gabrieldemarmiesse** [comment]
It's been a very long time since I made this repo, and most of the code wasn't written by me. If you want to make a PR I'll gladly merge it, but I don't think I'll take the time to dig into this repo again to make the modifications myself.

**Comment by scossu** [comment]
Thanks for responding. I wonder if this is affecting upstream too, but I can't test right now because I am exclusively working with Python 3.6.

All in all it's not a terrible issue, since the offending terms will pop up very soon in the C generation step.

**Comment by chrisjbillington** [comment]
I'm working on a fix for this, by appending underscores and including the C name as a string, which is how cython allows you to get around this issue when you're manually making pxd files.

So for example:
```
typedef int class;
void global(class from);
```
becomes
```
ctypedef int class_ 'class' 
cdef void global_ 'global'(class_ from_)
```

A remaining question I have is, should we escape names that are python/cython reserved words only, or should we also escape ones that shadow built in names, but are not actually syntax errors to include?

**Comment by chrisjbillington** [comment]
Decided it would be too intrusive to disallow shadowing builtins, it would come up very often. There are workarounds if shadowing a builtin, so I'll leave it. Whereas keywords are syntax errors.

**Comment by chrisjbillington** [comment]
Fixed by #9
